pkgs/development/libraries/libimobiledevice/default.nix

```diff
@@ -12,29 +12,24 @@
 , libimobiledevice-glue
 , SystemConfiguration
 , CoreFoundation
+, unstableGitUpdater
 }:
 
 stdenv.mkDerivation rec {
   pname = "libimobiledevice";
-  version = "1.3.0+date=2023-04-30";
+  version = "1.3.0-unstable-2024-05-20";
 
   outputs = [ "out" "dev" ];
 
+  passthru.updateScript = unstableGitUpdater { };
```

Comment:
`passthru` should be added just before `meta`.

There is a thing called `phases order` which is the order phases are executed.

See manual here:
https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases

Whenever adding attributes, consider where that attribute belongs to, and add at that point of derivation. Errors will appear in sequence when doing this. It's the order of execution.

Reply:
> Whenever adding attributes, consider where that attribute belongs to, and add at that point of derivation. Errors will appear in sequence when doing this. It's the order of execution.

I'm sorry, but I do not know which attribute belongs to what phase. Is there any documentation for that, other than searching through the entire nixpkgs source code each time?

Reply:
I just told you the answer.

Reply:
> `passthru` should be added just before `meta`
> I'd add `output` just before `passhtru`

so the `meta`, `output` and `passthru` are all in a phase after `configure` ?

Reply:
`preAutoreconf`, `configureFlags` should be before buildPhase, meaning before `nativeBuildInputs`.

Also skip a line between each of these attributes. It aids in reading.